download/client.go

```diff
@@ -92,13 +95,25 @@ func newClient() *http.Client {
 	}
 }
 
+// checkRedirectFunc is a wrapper around http.Client.CheckRedirect that allows for printing out redirects
 func checkRedirectFunc(req *http.Request, via []*http.Request) error {
 	if viper.GetBool(optname.Verbose) {
 		fmt.Printf("Received redirect '%d' to '%s'\n", req.Response.StatusCode, req.URL.String())
 	}
 	return nil
 }
 
+// transportDialContext is a wrapper around net.Dialer that allows for overriding DNS lookups via the values passed to
+// `--resolve` argument.
 func transportDialContext(dialer *net.Dialer) func(context.Context, string, string) (net.Conn, error) {
-	return dialer.DialContext
+	// Allow for overriding DNS lookups in the dialer without impacting Host and SSL resolution
+	return func(ctx context.Context, network, addr string) (net.Conn, error) {
+		if addrOverride := config.HostToIPResolutionMap[dialer.LocalAddr.String()]; addrOverride != "" {
```

Comment:
This looks wrong? I would expect HostToIPResolutionMap contains strings of the form `example.com:443` and dialer.LocalAddr.String() is of the form `0.0.0.0` or `127.0.0.1`.

I think we should be using `addr` here instead maybe?